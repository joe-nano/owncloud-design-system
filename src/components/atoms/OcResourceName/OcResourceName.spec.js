import { shallowMount } from "@vue/test-utils"

import Name from "./OcResourceName.vue"
import OcButton from "../../atoms/OcButton/OcButton.vue"

const fullPath = "images/nature/forest.jpg"
const name = "forest.jpg"
const extension = "jpg"
const type = "file"

const stubs = {
  "oc-button": OcButton,
  "oc-icon": true,
}

describe("OcResourceName", () => {
  it("doesn't add extension to hidden folder", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath: ".hidden",
        name: ".hidden",
        extension: "",
        type: "folder",
      },
      stubs,
    })

    expect(wrapper.find(".oc-resource-basename").text()).toMatch(".hidden")
    expect(wrapper.find(".oc-resource-extension").exists()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it("renders folder names with dots completely in the basename", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath: "folder.with.dots",
        name: "folder.with.dots",
        extension: "",
        type: "folder",
      },
      stubs,
    })

    expect(wrapper.find(".oc-resource-basename").text()).toMatch("folder.with.dots")
    expect(wrapper.find(".oc-resource-extension").exists()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it("has properties for resource path, name and type", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath,
        name,
        extension,
        type,
      },
      stubs,
    })

    const node = wrapper.find(".oc-resource-name")
    expect(node.attributes("data-test-resource-type")).toMatch(type)
    expect(node.attributes("data-test-resource-name")).toMatch(name)
    expect(node.attributes("data-test-resource-path")).toMatch("/" + name)
  })
  it("click on quick rename should trigger rename event", async () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath,
        name,
        extension,
        type,
      },
      stubs,
    })

    wrapper.find(".oc-resource-rename").trigger("click")
    await wrapper.vm.$nextTick()
    expect(wrapper.emitted().rename).toBeTruthy()
  })
})
