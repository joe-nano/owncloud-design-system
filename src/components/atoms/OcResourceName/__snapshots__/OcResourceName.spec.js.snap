// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcResourceName doesn't add extension to hidden folder 1`] = `
<span class="oc-resource-name-wrapper"><span data-test-resource-path=".hidden" data-test-resource-name=".hidden" data-test-resource-type="folder" class="oc-resource-name"><span class="oc-text-truncate"><span class="oc-resource-basename">.hidden</span></span>
<!----></span> <button type="button" class="oc-resource-rename oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw">
  <oc-icon-stub name="edit-2" fill-type="line" size="small"></oc-icon-stub>
</button></span>
`;

exports[`OcResourceName renders folder names with dots completely in the basename 1`] = `
<span class="oc-resource-name-wrapper"><span data-test-resource-path="folder.with.dots" data-test-resource-name="folder.with.dots" data-test-resource-type="folder" class="oc-resource-name"><span class="oc-text-truncate"><span class="oc-resource-basename">folder.with.dots</span></span>
<!----></span> <button type="button" class="oc-resource-rename oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw">
  <oc-icon-stub name="edit-2" fill-type="line" size="small"></oc-icon-stub>
</button></span>
`;
